internal/diff/diff.go

```diff
@@ -0,0 +1,133 @@
+package diff
+
+import (
+	"cmp"
+	"slices"
+)
+
+type diffResult struct {
+	Size     int64         `json:"size"`
+	Packages []diffPackage `json:"packages"`
+	Sections []diffSection `json:"sections"`
+}
+
+type changeType string
+
+const (
+	changeTypeAdd    changeType = "add"
+	changeTypeRemove changeType = "remove"
+	changeTypeChange changeType = "change"
+)
+
+type diffBase struct {
+	Name       string     `json:"name"`
+	From       int64      `json:"from"`
+	To         int64      `json:"to"`
+	ChangeType changeType `json:"change_type"`
+}
+
+func diffBaseCmp(a, b diffBase) int {
+	return -cmp.Compare(a.To-a.From, b.To-b.From)
+}
+
+type diffPackage struct {
+	diffBase
+}
+
+type diffSection struct {
+	diffBase
+	oldFileSize  int64
+	oldKnownSize int64
+	newFileSize  int64
+	newKnownSize int64
+}
+
+func processPackages(newPackages, oldPackages map[string]commonPackage) (ret []diffPackage) {
+	for k, v := range newPackages {
+		typ := changeTypeAdd
+		fromSize := int64(0)
+		if oldV, ok := oldPackages[k]; ok {
+			if v.Size == oldV.Size {
+				continue
+			}
+			typ = changeTypeChange
+			fromSize = oldV.Size
+		}
+		ret = append(ret, diffPackage{
+			diffBase: diffBase{Name: k, From: fromSize, To: v.Size, ChangeType: typ},
+		})
+	}
+
+	for k, v := range oldPackages {
+		if _, ok := newPackages[k]; !ok {
+			ret = append(ret, diffPackage{
+				diffBase: diffBase{Name: k, From: v.Size, To: 0, ChangeType: changeTypeRemove},
+			})
+		}
+	}
+
+	return
+}
+
+func processSections(newSections, oldSections []commonSection) (ret []diffSection) {
+	newSectionsMap := make(map[string]commonSection)
+	oldSectionsMap := make(map[string]commonSection)
+
+	for _, v := range newSections {
+		newSectionsMap[v.Name] = v
+	}
+	for _, v := range oldSections {
+		oldSectionsMap[v.Name] = v
+	}
+
+	for k, v := range newSectionsMap {
+		typ := changeTypeAdd
+		fromSize := int64(0)
+		fromFileSize := int64(0)
+		fromKnownSize := int64(0)
+		if oldV, ok := oldSectionsMap[k]; ok {
+			if v.UnknownSize() == oldV.UnknownSize() {
+				continue
+			}
+			typ = changeTypeChange
+			fromSize = oldV.UnknownSize()
+			fromFileSize = oldV.FileSize
+			fromKnownSize = oldV.KnownSize
+		}
+		ret = append(ret, diffSection{
+			diffBase:     diffBase{Name: k, From: fromSize, To: v.UnknownSize(), ChangeType: typ},
+			oldFileSize:  fromFileSize,
+			oldKnownSize: fromKnownSize,
+			newFileSize:  v.FileSize,
+			newKnownSize: v.KnownSize,
+		})
+	}
+
+	for k, v := range oldSectionsMap {
+		if _, ok := newSectionsMap[k]; !ok {
```

Comment:
```suggestion
		if _, ok := newSectionsMap[k]; ok {
     continue
}
```

Reply:
Packages not found in the new result are considered to be removed